### PR TITLE
Add Types for quicksettings

### PR DESCRIPTION
**Pull request by janizde**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## quicksettings ([<kbd>unpkg</kbd>](https://unpkg.com/browse/quicksettings@latest/))
was missing the following properties:
1. useExtStyleSheet
1. create
1. destroy
1. getValuesAsJSON
1. setValuesFromJSON

<details>
<summary>as well as these 58 other properties...</summary>
<p>saveInLocalStorage, clearLocalStorage, setPosition, setSize, setWidth, setHeight, setDraggable, setGlobalChangeHandler, hide, show, toggleVisibility, setCollapsible, collapse, expand, toggleCollapsed, setKey, removeControl, enableControl, disableControl, hideControl, showControl, overrideStyle, hideTitle, showTitle, hideAllTitles, showAllTitles, getValue, setValue, addBoolean, bindBoolean, addButton, addColor, bindColor, addDate, bindDate, addDropDown, bindDropDown, addElement, addFileChooser, addHTML, addImage, addRange, addNumber, bindRange, bindNumber, setRangeParameters, setNumberParameters, addPassword, bindPassword, addProgressBar, setProgressMax, addText, bindText, addTextArea, setTextAreaRows, bindTextArea, addTime, bindTime</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 26a12782eb28031c8d961e0c9000fa0562cd792e
</p>

File: types/quicksettings/quicksettings-tests.ts [hunk 1]
```diff
@@ -0,0 +1,746 @@
...
+qsAnyModel.addButton('foo', () => {}); // $ExpectType QuickSettingsPanel<Record<string, any>, string>
+qsTestModel.addButton('testStatic', () => {}); // $ExpectType QuickSettingsPanel<TestModel, "testStatic">
+qsTestModel.addButton('foo', () => {}); // $ExpectError
+qsTestModel.addButton('testString', () => {}); // $ExpectError
+
+// QuickSettingsPanel.addElement
+qsAnyModel.addElement('foo', document.createElement('div')); // $ExpectType QuickSettingsPanel<Record<string, any>, string>
+qsTestModel.addElement('testStatic', document.createElement('div')); // $ExpectType QuickSettingsPanel<TestModel, "testStatic">
+qsTestModel.addElement('foo', document.createElement('div')); // $ExpectError
+qsTestModel.addElement('testString', document.createElement('div')); // $ExpectError
```
Comment (orta) [inline types/quicksettings/quicksettings-tests.ts:746]: wowee, great work on these tests!

**Review by orta — APPROVED** [review]
Looks great, congrats!

File: types/quicksettings/quicksettings-tests.ts (hunk 1 above)
Comment (janizde) [reply types/quicksettings/quicksettings-tests.ts:746]: Thanks for the review!

**Comment by janizde** [comment]
Ready to merge

**Pull request merged**